config/application.rb

```diff
@@ -1,17 +1,19 @@
-require_relative "boot"
+# frozen_string_literal: true
 
-require "rails"
+require_relative 'boot'
+
+require 'rails'
 # Pick the frameworks you want:
```

Comment:
- [ ] Good job adding the `rack-cors` gem in the gemfile but you didn't add the code to handle the `CORS policy`. Please make sure to fix this issue.

![image](https://user-images.githubusercontent.com/83803414/211802005-b5aadcf2-b505-4b00-9314-21ed7e86e0cb.png)

  - To fix this issue add the below code in the application class of this file.

```ruby
config.middleware.insert_before 0, Rack::Cors do
  allow do
    origins /\Ahttp:\/\/localhost:\d+\z/
    resource '*', headers: :any, methods: :any
  end
end

Reply:
@thecodechaser Thank you for taking the time to leave this excellent review.

From my frontend code, I have this:
 
// Api
const url = 'http://127.0.0.1:4000/api/v1/random-greeting';

From the URL above the port is 4000, and the `origins "http://localhost:3000"` from my frontend is allowed in `cors.rb` and everything works fine on my end.

I'm wondering why you have an `api` request to `port 3000` from your `screenshot` and yet I'm not calling that on the frontend. If you have a clue why this is happening kindly share it with me.

Reply:
![cors](https://user-images.githubusercontent.com/46409424/211835589-3e6cc8c6-c031-4940-8806-b0e246422cc7.png)

Reply:
@alphayowakarindi Hi, On my machine, The port is running on `3000`, not on `4000`, and your code only allows `3000` port. I think this is the reason why the error message was there.
I would suggest you use the origin `origins /\Ahttp:\/\/localhost:\d+\z/ in your cors.rb file. It will allow all origins to access the API.

Reply:
@thecodechaser I'll make the change.

My goal was for the `api` to be consumed by my fronted domain only and that's the reason I specified in the README.md that the backend has to run on port `4000` for it to work. 

What's your opinion on that approach?

Reply:
@thecodechaser Okay, thank you.